web/src/ferrostar-core.ts

```diff
@@ -121,23 +144,16 @@ class FerrostarCore extends LitElement {
     this.currentLocationMapMarker = L.marker(route.geometry[0]).addTo(this.map!);
   }
 
-  async replaceRoute(route: any, config: any) {
-    // TODO
-  }
-
-  async advanceToNextStep() {
-    // TODO
-  }
-
   async stopNavigation() {
-    // TODO
-  }
-
-  private async handleStateUpdate(newState: any, location: any) {
-    // TODO
+    this.resetMap();
```

Comment:
Jacob and I noticed that this is doing a lot of mixing of UI and core layers. This doesn't need to happen in this PR, but we should probably factor this out at some point. We can brainstorm ideas for this on a call later.

```suggestion
    // TODO: Factor out the UI layer from the core
    this.resetMap();
```

Reply:
I agree that this approach is temporary; once we move to MapLibre we should rewrite these.